client/controller/index.web.js

```diff
@@ -393,11 +393,27 @@ export const notFound = ( context, next ) => {
 /**
  * Middleware to set the selected site ID based on the `origin_site_id` query parameter.
  */
-export const setSelectedSiteIdByOrigin = ( context, next ) => {
+export const setGloballySelectedSiteIdByOrigin = ( context, next ) => {
+	/**
+	 * The `origin_site_id` query parameter overrides the selected site ID.
+	 * This param is passed from wp-admin.
+	 */
 	const originSiteId = ( context.query.origin_site_id ?? '' ).trim();
 	if ( originSiteId ) {
-		context.store.dispatch( setSelectedSiteId( originSiteId ) );
+		context.store.dispatch( setGloballySelectedSiteId( originSiteId ) );
 		context.page.replace( removeQueryArgs( context.canonicalPath, 'origin_site_id' ) );
+		next();
+		return;
 	}
+
+	/**
+	 * Set selected site ID globally to use it on /domains/*, /themes, and /plugins, as well,
+	 * since The selected site ID is cleared on on /domains/*, /themes, /plugins, as those pages are aimed for non-site-specific context.
+	 */
+	const selectedSiteId = getSelectedSiteId( context.store.getState() );
+	if ( selectedSiteId ) {
+		context.store.dispatch( setGloballySelectedSiteId( selectedSiteId ) );
```

Comment:
Im a little concerned with managing a separate siteId that we have to dispatch changes for. I think it opens up potential edge cases where it might get out of sync from where we want, if we have to dispatch to change it every time it needs to change.

Alternatively, I like what you already have from your separate PRs to set the selectedSiteId here instead. My idea for a `prevSelectedSiteId` I am going to look at is more along the idea of something we never have to dispatch or manage on our own. We just build this into the state so it is automagically updated whenever selectedSiteId is updated, and always in sync with what we need without us ever having to dispatch an action from outside the store.

Reply:
> My idea for a prevSelectedSiteId I am going to look at

Implemented here https://github.com/Automattic/wp-calypso/pull/92800